### ISSUE TITLE
Double-check whether predicate expressions have correct nullability semantics

### ISSUE THREAD

**Issue by jvanstraten** [substrait-io/substrait-validator#51]
When I wrote the bulk of the validation code, I was properly confused about nullable booleans in the context of predicate expressions (both things that want one, and things that return one, like set containment subqueries). There is a good summary about what the behavior of these things *should* be [here](https://github.com/substrait-io/substrait/issues/325#issuecomment-1244578057), but it's probably worth double-checking corner cases against query engine documentation or just by experimentation. When a nullable boolean is received by a predicate expression slot, the validator should probably also emit a note of some kind about what null means in that particular context.

**Comment by mbasmanova** [comment]
@jvanstraten Curious what is the "validation code"? Is there some sort of a test that specifies query plans and inputs and expected outputs? I'd be curious to see what we have for the joins.

**Comment by jvanstraten** [comment]
The validator (this repository) only checks whether a plan is valid, and sort of describes in English how the individual relations and expressions should behave; it doesn't know how to actually execute a plan given some input tables. The descriptions are probably not precise enough for joins in this case, because all the descriptions are currently limited to my own personal understanding. I would *love* some double-checking of the descriptions it does emit, though. Running the validator should be super easy if you have some plans already (see toplevel readme file); if you disagree with something it says or if you think the spec disagrees with it, please open an issue here.

**Comment by mbasmanova** [comment]
@jvanstraten This sounds super interesting. I'd love to try it out. Did you mean to include a link for "(this repository)"? For some reason it shows up as regular text.

**Comment by mbasmanova** [comment]
Oh... perhaps, you literally meant this repository: https://github.com/substrait-io/substrait-validator

Let me try it.

**Comment by jvanstraten** [comment]
Yeah, that's exactly what this issue is for. @mbrobbel  is already working on it.

**Comment by mbasmanova** [comment]
@jvanstraten Great! Thank you for clarifying.

**Comment by mbrobbel** [comment]
@jvanstraten you can assign this one to me.
